src/test/codewhisperer/util/authUtil.test.ts

```diff
@@ -0,0 +1,180 @@
+    return {
+        type: 'sso',
+        ssoRegion: 'us-east-1',
+        startUrl: builderIdStartUrl,
+        ...props,
+    }
+}
+
+function createEntSsoProfile(props?: Partial<Omit<SsoProfile, 'type'>>): SsoProfile {
+    return {
+        type: 'sso',
+        ssoRegion: 'us-east-1',
+        startUrl: enterpriseSsoStartUrl,
+        ...props,
+    }
+}
+
+const tokenProviders = new Map<string, ReturnType<typeof createTestTokenProvider>>()
+
+function createTestTokenProvider() {
+    let token: SsoToken | undefined
+    const provider = stub(SsoAccessTokenProvider)
+    provider.getToken.callsFake(async () => token)
+    provider.createToken.callsFake(
+        async () => (token = { accessToken: '123', expiresAt: new Date(Date.now() + 1000000) })
+    )
+    provider.invalidate.callsFake(async () => (token = undefined))
+
+    return provider
+}
+
+function getTestTokenProvider(...[profile]: ConstructorParameters<typeof SsoAccessTokenProvider>) {
+    const key = getSsoProfileKey(profile)
+    const cachedProvider = tokenProviders.get(key)
+    if (cachedProvider !== undefined) {
+        return cachedProvider
+    }
+
+    const provider = createTestTokenProvider()
+    tokenProviders.set(key, provider)
+
+    return provider
+}
+
+let auth: Auth
+let store: ProfileStore
+let builderIdConn: SsoConnection | undefined
+let ssoConn: SsoConnection
+let entSsoConn: SsoConnection
+let mockConnListNoBuilder: Connection[]
+let mockConnListNoEntSso: Connection[]
+const ssoProfile = createSsoProfile()
+const awsBuilderIdProfile = createAwsBuilderIdProfile({ scopes: codewhispererScopes })
+const enterpriseSsoProfile = createEntSsoProfile({ scopes: codewhispererScopes })
+
+async function createMockConnections(){
+    store = new ProfileStore(new FakeMemento())
+    auth = new Auth(store, getTestTokenProvider, new CredentialsProviderManager())
+    builderIdConn = await auth.createConnection(awsBuilderIdProfile)
+    ssoConn = await auth.createConnection(ssoProfile)
+    entSsoConn = await auth.createConnection(enterpriseSsoProfile)
+}
+
+describe('AuthUtil', async function () {
+   before(async function () {
+       await createMockConnections()
+   })
+
+    beforeEach(async function () {
+        sinon.stub(Commands, 'register') 
+        store = new ProfileStore(new FakeMemento())
+        auth = new Auth(store, getTestTokenProvider, new CredentialsProviderManager())
+        
+        sinon.replace(SsoClient, 'create', () => {
+            const s = stub(SsoClient)
+            s.logout.resolves()
+
+            return s
+        })
+              
+    })
+
+    afterEach(async function () {
+        sinon.restore()
+    })
+
+    //TODO
+    //factory function to create connectionlists dynamically
+    //stub secondaryAuth useNewConnection? vs auth instance use connection
+    //remove auth and profile from beforeEach and see if it breaks
+    //todo: figure out what sinon.replace in beforeEach does
```

Comment:
Oh it just removes noisy error messages from the test output when calling `Auth.logout`